test/insertion_order_parser_test.dart

```diff
@@ -0,0 +1,302 @@
+import 'package:dv360_excel_plugin/src/proto/insertion_order.pb.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group(InsertionOrderParser, () {
+    String input;
+    const emptyEntry = '';
+
+    InsertionOrder insertionOrderTemplate;
+    InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+        oneDateRange;
+    InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+        oneStartDate;
+    InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+        oneEndDate;
+
+    setUp(() {
+      insertionOrderTemplate = InsertionOrder()
+        ..advertiserId = emptyEntry
+        ..campaignId = emptyEntry
+        ..insertionOrderId = emptyEntry
+        ..displayName = emptyEntry
+        ..updateTime = emptyEntry
+        ..entityStatus = InsertionOrder_EntityStatus.ENTITY_STATUS_UNSPECIFIED
+        ..pacing = InsertionOrder_Pacing()
+        ..budget = InsertionOrder_InsertionOrderBudget();
+
+      oneStartDate =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+            ..year = 2019
+            ..month = 1
+            ..day = 1;
+
+      oneEndDate =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+            ..year = 2019
+            ..month = 12
+            ..day = 31;
+
+      oneDateRange =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+            ..startDate = oneStartDate
+            ..endDate = oneEndDate;
+    });
+
+    tearDown(disposeAnyRunningTest);
+
+    test(
+        'parse insertionOrder from empty json should return '
+        'an empty insertionOrder instance', () {
+      input = '{}';
+
+      final actual = InsertionOrderParser.parse(input);
+
+      final expected = InsertionOrder();
+      expect(actual, expected);
+    });
+
+    test(
+        'parse insertionOrder from json that contains only advertiserId '
+        'should return an instance with only advertiserId set', () {
+      input = '{"advertiserId":"111111"}';
+
+      final actual = InsertionOrderParser.parse(input);
+
+      final expected = insertionOrderTemplate..advertiserId = '111111';
+      expect(actual, expected);
+    });
+
+    test('parse insertionOrder from json that contains everything', () {
+      final advertiserId = '"advertiserId":"11111"';
+      final campaignId = '"campaignId":"2222222"';
+      final insertionOrderId = '"insertionOrderId":"3333333"';
+      final displayName = '"displayName":"display name"';
+      final entityStatus = '"entityStatus":"ENTITY_STATUS_ACTIVE"';
+      final updateTime = '"updateTime":"2020-06-23T17:14:58.685Z"';
+      final pacing = '"pacing":{"pacingPeriod":"PACING_PERIOD_FLIGHT",'
+          '"pacingType":"PACING_TYPE_AHEAD"}';
+      final budgetSegment = '"budgetSegments":['
+          '{"budgetAmountMicros":"4000000","description":"year-2019",'
```

Comment:
For multiline string constants you can use the triple quotes like this:
final budgetSegment = """
   "budgetSegments": [
       {"budgetAmountMicros":....
""";